src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/DateTimeVariantTest.cs

```diff
@@ -73,55 +88,48 @@ private static void SendInfo(object paramValue, string expectedTypeName, string
-            DisplayHeader(tag, paramValue, expectedBaseTypeName);
             string procName = DataTestUtility.GetUniqueNameForSqlServer("paramProc1");
             try
             {
-                using (SqlConnection conn = new SqlConnection(s_connStr))
-                {
-                    conn.Open();
-                    DropStoredProcedure(conn, procName);
-                    xsql(conn, string.Format("create proc {0} (@param {1}) as begin select @param end;", procName, expectedBaseTypeName));
+                using SqlConnection conn = new SqlConnection(s_connStr);
+                conn.Open();
+                DropStoredProcedure(conn, procName);
+                xsql(conn, string.Format("create proc {0} (@param {1}) as begin select @param end;", procName, expectedBaseTypeName));
 
-                    using (SqlCommand cmd = conn.CreateCommand())
-                    {
-                        cmd.CommandText = procName;
-                        cmd.CommandType = CommandType.StoredProcedure;
-                        SqlParameter p = cmd.Parameters.AddWithValue("@param", paramValue);
-                        cmd.Parameters[0].SqlDbType = GetSqlDbType(expectedBaseTypeName);
-                        using (SqlDataReader dr = cmd.ExecuteReader())
-                        {
-                            dr.Read();
-                            VerifyReaderTypeAndValue("Test Simple Parameter [Data Type]", expectedBaseTypeName, expectedTypeName, dr[0], expectedTypeName, paramValue);
-                            dr.Dispose();
-                        }
-                    }
-                }
+                using SqlCommand cmd = conn.CreateCommand();
+                cmd.CommandText = procName;
+                cmd.CommandType = CommandType.StoredProcedure;
+
+                SqlParameter p = cmd.Parameters.AddWithValue("@param", paramValue);
+                cmd.Parameters[0].SqlDbType = GetSqlDbType(expectedBaseTypeName);
+
+                using SqlDataReader dr = cmd.ExecuteReader();
+                dr.Read();
+                VerifyReaderTypeAndValue("Test Simple Parameter [Data Type]", expectedBaseTypeName, expectedTypeName, dr[0], expectedTypeName, paramValue);
+                dr.Dispose();
             }
             catch (Exception e)
             {
-                if (IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName))
-                    LogMessage(tag, "[EXPECTED EXCEPTION] " + e.Message);
-                else
-                    DisplayError(tag, e);
+                // Invalid cast exception is also permited here to prevent failure as the caller is waiting for InvalidCastException
+                Assert.True(IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName) || e is InvalidCastException, $"[UNEXPECTED EXCEPTION]: {e.Message}");
+
+                // Exception is thrown for the caller method which is expecting invalid cast exception at line 58.
```

Comment:
This comment will most likely become out-of-date. We can just omit "at line 58".